### PR TITLE
Add lookupsWithBlobs and use that in tests

### PR DESCRIPTION
**Pull request by phadej**
As Arbitrary Updates already insert blobs,
we get a bit of blob lookup testing out of this.

### CODE REVIEW

**Review by jorisdral — APPROVED** [review]
LGTM

File: test/Test/Database/LSMTree/ModelIO/Normal.hs [hunk 1]
```diff
@@ -48,6 +51,33 @@ makeNewTable h ups = do
...
 
+-- | Like 'retrieveBlobs' but works for any 'Traversable'.
+-- 
+-- Like 'partsOf' in @lens@ this uses state monad.
+retrieveBlobsTrav ::
+  (IsTableHandle h, IOLike m, SomeSerialisationConstraint blob, Traversable t)
+  => h m k v blob -> t (BlobRef h blob) -> m (t blob)
+retrieveBlobsTrav hdl brefs = do
+  blobs <- retrieveBlobs hdl (toList brefs)
+  evalStateT (traverse (\_ -> state un) brefs) blobs
```
Comment (dcoutts) [inline test/Test/Database/LSMTree/ModelIO/Normal.hs:62]: it took me a while to figure out what this is doing

**Review by dcoutts — APPROVED** [review]

File: test/Test/Database/LSMTree/ModelIO/Normal.hs (hunk 1 above)
Comment (phadej) [reply test/Test/Database/LSMTree/ModelIO/Normal.hs:62]: ... but it says it in the comment.

Comment (phadej) [reply test/Test/Database/LSMTree/ModelIO/Normal.hs:63]: longer explanation https://oleg.fi/gists/posts/2023-10-12-use-traversals-for-batch-operations.html

**Pull request merged**